homeassistant/components/ituran/sensor.py

```diff
@@ -0,0 +1,111 @@
+"""Sensors for Ituran vehicles."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.const import (
+    DEGREE,
+    UnitOfElectricPotential,
+    UnitOfLength,
+    UnitOfSpeed,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import IturanConfigEntry
+from .coordinator import IturanDataUpdateCoordinator
+from .entity import IturanBaseEntity
+
+
+@dataclass(frozen=True)
+class IturanSensorEntityDescription(SensorEntityDescription):
+    """Describes Ituran sensor entity."""
+
+    enabled_by_default: bool = True
+
+
+SENSOR_TYPES: list[IturanSensorEntityDescription] = [
+    IturanSensorEntityDescription(
+        key="address",
+        translation_key="address",
+        enabled_by_default=False,
+    ),
+    IturanSensorEntityDescription(
+        key="battery_voltage",
+        translation_key="battery_voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=UnitOfElectricPotential.VOLT,
+        suggested_display_precision=0,
+        enabled_by_default=False,
+    ),
+    IturanSensorEntityDescription(
+        key="heading",
+        translation_key="heading",
+        native_unit_of_measurement=DEGREE,
+        suggested_display_precision=0,
+        enabled_by_default=False,
+    ),
+    IturanSensorEntityDescription(
+        key="last_update",
+        translation_key="last_update",
+        device_class=SensorDeviceClass.TIMESTAMP,
+        enabled_by_default=False,
+    ),
+    IturanSensorEntityDescription(
+        key="mileage",
+        translation_key="mileage",
+        device_class=SensorDeviceClass.DISTANCE,
+        native_unit_of_measurement=UnitOfLength.KILOMETERS,
+        suggested_display_precision=2,
+    ),
+    IturanSensorEntityDescription(
+        key="speed",
+        translation_key="speed",
```

Comment:
this one can take it's translation and icon from the device class if you leave the translation key on `None` (or don't set it)

Reply:
Removed